packages/wallet/src/util.ts

```diff
@@ -1 +1,158 @@
+  const logger = props.logger || dummyLogger;
+  return {
+    getBalance: async (): Promise<Cbor> => {
+      logger.debug('getting balance');
+      try {
+        const value = await firstValueFrom(wallet.balance.available$);
+        return Buffer.from(coreToCsl.value(value).to_bytes()).toString('hex');
+      } catch (error) {
+        logger.error(error);
+        throw error;
+      }
+    },
+    getChangeAddress: async (): Promise<Cbor> => {
+      logger.debug('getting changeAddress');
+      try {
+        const [{ address }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(address as unknown as string);
+
+        if (!address || !parsedAddress) {
+          logger.error('could not get change address');
+          throw new ApiError(500, 'could not get change address');
+        } else {
+          return Buffer.from(parsedAddress.to_bytes()).toString('hex');
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getNetworkId: async (): Promise<number> => {
+      logger.debug('getting networkId');
+      return Promise.resolve(props.networkId);
+    },
+    getRewardAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting reward addresses');
+      try {
+        const [{ rewardAccount }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(rewardAccount as unknown as string);
+
+        if (!rewardAccount || !parsedAddress) {
+          throw new ApiError(500, 'could not get reward address');
+        } else {
+          return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getUnusedAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting unused addresses');
+      return Promise.resolve([]);
+    },
+    getUsedAddresses: async (_paginate?: Paginate): Promise<Cbor[]> => {
+      logger.debug('getting changeAddress');
+
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address as unknown as string);
+
+      if (!address || !parsedAddress) {
+        throw new ApiError(500, 'could not get used addresses');
+      } else {
+        return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+      }
+    },
+    getUtxos: async (amount?: Cbor, paginate?: Paginate): Promise<Cardano.Utxo[] | undefined> => {
+      const utxos = await firstValueFrom(wallet.utxo.available$);
+
+      // eslint-disable-next-line unicorn/no-useless-undefined
+      if (!utxos) return Promise.resolve(undefined);
+      const cslUtxos = coreToCsl.utxo(utxos);
+
+      if (amount) {
+        try {
+          const filterAmount = CSL.Value.from_bytes(Buffer.from(amount, 'hex'));
+          const filteredUtxosByAmount = cslUtxos.filter(
```

Comment:
An easy way to get total value of utxo is `Cardano.util.coalesceValueQuantities(utxo.map(([_, txOut]) => txOut.value))`